data-pipeline-ffi/src/span.rs

```diff
@@ -0,0 +1,1009 @@
+        object.$field.remove(&bytes_str_key);
+    }};
+}
+
+macro_rules! exists_hashmap {
+    ($ptr:expr, $key:expr, $field:ident) => {{
+        if $ptr.is_null() {
+            return false;
+        }
+        let object = &mut *$ptr;
+        let bytes_str_key = BytesString::from_slice($key.as_bytes()).unwrap_or_default();
+        return object.$field.contains_key(&bytes_str_key);
+    }};
+}
+
+macro_rules! get_keys_hashmap {
+    ($span_ptr:expr, $out_count:expr, $field:ident) => {{
+        if $span_ptr.is_null() || $out_count.is_null() {
+            return std::ptr::null_mut();
+        }
+
+        let span = &mut *$span_ptr;
+
+        let mut slices: Vec<CharSlice<'static>> = Vec::with_capacity(span.$field.len());
+
+        for key in span.$field.keys() {
+            let value = CharSlice::from_raw_parts(key.as_str().as_ptr().cast(), key.as_str().len());
+            slices.push(value);
+        }
+
+        let slice_box = slices.into_boxed_slice();
+        *$out_count = slice_box.len();
+
+        Box::into_raw(slice_box) as *mut CharSlice<'static>
+    }};
+}
+
+macro_rules! new_vector_item {
+    ($ptr:expr, $field:ident, $item_type:ty) => {{
+        if $ptr.is_null() {
+            return std::ptr::null_mut();
+        }
+
+        let object = unsafe { &mut *$ptr };
+        object.$field.push(<$item_type>::default());
+
+        let item = object.$field.last_mut().unwrap();
+
+        item as *mut $item_type
+    }};
+}
+
+macro_rules! set_event_attribute {
+    ($ptr:expr, $key:expr, $new_item:expr) => {{
+        if $ptr.is_null() {
+            return;
+        }
+
+        let event = unsafe { &mut *$ptr };
+        let key = BytesString::from_slice($key.as_bytes()).unwrap_or_default();
+        let value = $new_item;
+
+        // Remove previous value if it exists
+        let previous = event.attributes.remove(&key);
+
+        // Merge the previous value with the new one
+        let merged = match previous {
+            None => AttributeAnyValueBytes::SingleValue(value),
+            Some(AttributeAnyValueBytes::SingleValue(x)) => {
+                AttributeAnyValueBytes::Array(vec![x, value])
+            }
+            Some(AttributeAnyValueBytes::Array(mut array)) => {
+                array.push(value);
+                AttributeAnyValueBytes::Array(array)
+            }
+        };
+
+        // Insert the merged value back into the map
+        event.attributes.insert(key, merged);
+    }};
+}
+
+// ------------------ TracesBytes ------------------
+
+pub type TraceBytes = Vec<SpanBytes>;
+pub type TracesBytes = Vec<TraceBytes>;
+
+#[no_mangle]
+pub extern "C" fn ddog_get_traces() -> *mut TracesBytes {
+    Box::into_raw(Box::default())
+}
+
+#[no_mangle]
+#[allow(clippy::missing_safety_doc)]
+pub unsafe extern "C" fn ddog_free_traces(ptr: *mut TracesBytes) {
+    if !ptr.is_null() {
+        drop(Box::from_raw(ptr));
+    }
+}
```

Comment:
```suggestion
pub unsafe extern "C" fn ddog_free_traces(ptr: Box<TracesBytes>) {
}
```
This is easier and will do the same (albeit no null-check, but you are not supposed to pass nullable values to stuff accepting Box<>)